### ISSUE TITLE
log failed attempts

### ISSUE THREAD

**Issue by jknockaert** [nextcloud/twofactor_totp#165]
I suggest that the app logs failed attempts much like the nextcloud core logs failed logins. That would allow to handle these security issues with external tools like fail2ban.

**Comment by ChristophWurst** [comment]
Yeah that would make sense. Since that also makes sense for the other 2FA providers, I suggest to reopen this enhancement ticket in the server repo. Please mention me there and cross-reference this ticket.

Implementing the logging should be easy. Let me know if you'd like to help with this enhancement and I can point you to the effected files. This would be a great starting point for a contribution to the Nextcloud project 😉 

On a side note: we do show failed 2FA attempts in the activity stream of the activity app on Nextcloud 12 and newer :-) ref #96

**Comment by ChristophWurst** [comment]
(I'll leave this ticket open until we have the proper replacement in the server repo. This one can be closed then)
